sdk/include/compartment-macros.h

```diff
@@ -5,33 +5,78 @@
+                                                name,                          \
+                                                mangledName,                   \
+                                                permitLoad,                    \
+                                                permitStore,                   \
+                                                permitLoadStoreCapabilities,   \
+                                                permitLoadMutable)             \
 	({                                                                         \
 		volatile type *ret; /* NOLINT(bugprone-macro-parentheses) */           \
-		__asm(".ifndef __import_mem_" #name "\n"                               \
-		      "  .type     __import_mem_" #name ",@object\n"                   \
+		__asm(".ifndef " mangledName "\n"                                      \
+		      "  .type     " mangledName ",@object\n"                          \
 		      "  .section  .compartment_imports." #name                        \
 		      ",\"awG\",@progbits," #name ",comdat\n"                          \
-		      "  .globl    __import_mem_" #name "\n"                           \
-		      "  .p2align  3\n"                                                \
-		      "__import_mem_" #name ":\n"                                      \
+		      "  .globl    " mangledName "\n"                                  \
+		      "  .p2align  3\n" mangledName ":\n"                              \
 		      "  .word __export_mem_" #name "\n"                               \
-		      "  .word __export_mem_" #name "_end - __export_mem_" #name "\n"  \
-		      "  .size __import_mem_" #name ", 8\n"                            \
+		      "  .word __export_mem_" #name "_end - __export_mem_" #name       \
+		      " + %c1\n"                                                       \
+		      "  .size " mangledName ", 8\n"                                   \
 		      " .previous\n"                                                   \
 		      ".endif\n"                                                       \
 		      "1:"                                                             \
 		      "  auipcc  %0,"                                                  \
-		      "      %%cheri_compartment_pccrel_hi(__import_mem_" #name ")\n"  \
+		      "      %%cheri_compartment_pccrel_hi(" mangledName ")\n"         \
 		      "  clc     %0, %%cheri_compartment_pccrel_lo(1b)(%0)\n"          \
-		      : "=C"(ret));                                                    \
+		      : "=C"(ret)                                                      \
+		      : "i"(((permitLoad) ? (1 << 31) : 0) +                           \
+		            ((permitStore) ? (1 << 30) : 0) +                          \
+		            ((permitLoadStoreCapabilities) ? (1 << 29) : 0) +          \
+		            ((permitLoadMutable) ? (1 << 28) : 0)));                   \
```

Comment:
Can you use symbolic constants here?

```suggestion
		      : "i"(((permitLoad) ? (1 << ImportEntry::PermitLoad) : 0) +      \
		            ((permitStore) ? (1 << ImportEntry::PermitStore) : 0) +    \
		            ((permitLoadStoreCapabilities) ? (1 << ImportEntry::PermitLoadStoreCapabilities) : 0) + \
		            ((permitLoadMutable) ? (1 << ImportEntry::PermitLoadMutable) : 0))); \
```

If this needs to be pure C, can you move those constant definitions out of `ImportEntry` and make that refer to the new definition site?

Reply:
It needs to be pure C, and public headers should not include private headers from the loader.  

This is part of the ABI and we have tests for it, so I'm not too worried about things being mismatched.  Tests will fail if these constants are wrong.

Reply:
I also don't want anything else using these constants and, at some point, this logic will all go into the compiler, so I don't want a symbolic constant to suggest to users that this is something that they should actually depend on.